packages/components/src/internal/util/WindowFocusCheckExpiredSession.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { FC, useEffect } from 'react';
+
+import { useServerContext } from '../components/base/ServerContext';
+
+export const WindowFocusCheckExpiredSession: FC = () => {
+    const { WebSocket } = useServerContext();
+    const onTabFocus = WebSocket?.checkForExpiredSession;
+
+    const onTabFocus_ = (): void => {
+        // Note the extra second (1s in this case) for the timeout before we query whoami, this
+        // is to allow time for the server login from the other tab to take hold.
+        setTimeout(() => {
+            onTabFocus();
```

Comment:
It looks like `onTabFocus` could be `undefined`. Should account for this with either `onTabFocus?.()` or better yet not wiring up any of the eventing in the case that it is not defined.

Reply:
Yup, good catch. I have the `if (!onTabFocus) return;` check before adding the event listener inside of the `useEffect`. I believe that should cover what you are describing.